src/wp-includes/media.php

```diff
@@ -1958,6 +1971,104 @@ function wp_img_tag_add_loading_attr( $image, $context ) {
+	$height             = preg_match( '/ height="([0-9]+)"/', $image, $match_height ) ? (int) $match_height[1] : '';
+	$optimization_attrs = wp_get_loading_optimization_attributes(
+		'img',
+		array(
+			'width'  => $width,
+			'height' => $height,
+		),
+		$context
+	);
+
+	/**
+	 * Filters the loading optimization attributes to be added to an image.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param string|bool $optimization_attrs   Attribute and it's values obtained from `wp_get_loading_optimization_attributes` function.
+	 * @param string      $image   The HTML `img` tag to be filtered.
+	 * @param string      $context Additional context about how the function was called or where the img tag is.
+	 */
+	$optimization_attrs = apply_filters( 'wp_img_tag_add_loading_optimization_attrs', $optimization_attrs, $image, $context );
+
+	/**
+	 * Filters the `loading` attribute value to add to an image. Default `lazy`.
+	 * This filter is added in for backward compatibility.
+	 *
+	 * Returning `false` or an empty string will not add the attribute.
+	 * Returning `true` will add the default value.
+	 * `true` and `false` usage supported for backward compatibility.
+	 *
+	 * @param string|bool $optimization_attrs['loading'] Current value for `loading` attribute for the image.
+	 * @param string      $image   The HTML `img` tag to be filtered.
+	 * @param string      $context Additional context about how the function was called or where the img tag is.
+	 */
+	$optimization_attrs['loading'] = apply_filters(
+		'wp_img_tag_add_loading_attr',
+		isset( $optimization_attrs['loading'] ) ? $optimization_attrs['loading'] : false,
+		$image,
+		$context
+	);
+
+	/**
+	 * Validate the values after filtering.
+	 */
+	if ( isset( $optimization_attrs['loading'] ) && $optimization_attrs['loading'] ) {
+		if ( ! in_array( $optimization_attrs['loading'], array( 'lazy', 'eager' ), true ) ) {
+			$optimization_attrs['loading'] = 'lazy';
+		}
+	}
+	if ( isset( $optimization_attrs['fetchpriority'] ) ) {
+		if ( ! in_array( $optimization_attrs['loading'], array( 'high', 'low', 'auto' ), true ) ) {
+			// For invalid values unset the attribute.
+			unset( $optimization_attrs['fetchpriority'] );
+		}
+	}
+
+	// Lazy-loading and fetchpriority='high' can't coexist at same time.
+	if (
+		isset( $optimization_attrs['loading'] ) && isset( $optimization_attrs['fetchpriority'] ) &&
+		'lazy' === $optimization_attrs['loading'] && 'high' === $optimization_attrs['fetchpriority']
+	) {
+		_doing_it_wrong(
+			__METHOD__,
+			__( 'An image cannot be lazy-loaded and assigned fetchpriority="high" at the same time.' ),
+			'6.3.0'
+		);
+		return $image;
+	}
+
+	// Retained for backward compatibility.
+	$add_img_loading_att = wp_lazy_loading_enabled( 'img', $context );
+
+	if (
+		$add_img_loading_att &&
+		false === strpos( $image, ' loading=' ) &&
+		isset( $optimization_attrs['loading'] ) && $optimization_attrs['loading']
```

Comment:
Why not simplify like you have below?
```suggestion
		! empty( $optimization_attrs['loading'] )
```

Reply:
@felixarntz It was my fault, i was in a mindset that empty() wont be true if `$optimization_attrs['loading']=false`